applicationinsights-react-native/Tests/Unit/src/reactnativeplugin.tests.ts

```diff
@@ -12,33 +12,90 @@ export class ReactNativePluginTests extends AITestClass {
     private config: IReactNativePluginConfig;
     private item: ITelemetryItem;
     private deviceModule: DeviceModule;
+    private coreConfig: IConfiguration;
 
     public testInitialize() {
         this._disableDynProtoBaseFuncs();
         this.core = new AppInsightsCore();
         this.core.logger = new DiagnosticLogger();
         this.plugin = new ReactNativePlugin();
         this.deviceModule = new DeviceModule();
-        this.plugin.setDeviceInfoModule(this.deviceModule)
+        this.plugin.setDeviceInfoModule(this.deviceModule);
         this.config = {};
+        this.coreConfig = {
+            instrumentationKey: 'testIkey',
+            endpointUrl: 'testEndpoint',
+            extensionConfig: {}
+        };
     }
 
     public testCleanup() {
-        this.core = null;
-        this.plugin = null;
-        this.config = null;
+        if (this.core && this.core.isInitialized()) {
```

Comment:
I think we should still be resetting the core, plugin and config to null after unloading so that we cleanup the previous values.